js/main.js

```diff
@@ -29,58 +29,145 @@ Site = {
 };
 
 Site.Map = {
-  map,
+  map: null,
+  pos: {
+    lat: 0,
+    long: 0
+  },
+  coordinateArray: [],
 
   init: function() {
     var _this = this;
 
-    navigator.geolocation.getCurrentPosition(function(position) {
-      _this.initMap(position.coords.latitude, position.coords.longitude);
-    });
+    _this.map = L.map('map').setView([0, 0], 2);
+
+    _this.tileMap();
+
+    if (navigator.geolocation) {
+
+      _this.flyToDevicePos();
+
+    } else {
+      alert('your browser sucks');
+    }
   },
 
-  initMap: function(lat, long) {
+  tileMap: function() {
     var _this = this;
 
-    _this.map = L.map('map', {
-      dragging: false
-    }).setView([lat, long], 20);
-
-    L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpandmbXliNDBjZWd2M2x6bDk3c2ZtOTkifQ._QA7i5Mpkd_m30IGElHziw', {
+    var tileLayer = L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token=pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpandmbXliNDBjZWd2M2x6bDk3c2ZtOTkifQ._QA7i5Mpkd_m30IGElHziw', {
       maxZoom: 20,
       id: 'mapbox.streets'
     }).addTo(_this.map);
+  },
 
-    var coordinateArray = [];
+  getDevicePos: function(callback) {
+    var _this = this;
 
-    _this.map.on('mouseup', function() {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      _this.pos.lat = position.coords.latitude;
+      _this.pos.long = position.coords.longitude;
 
-      _this.map.off("mousemove");
+      callback();
+    });
+  },
 
-      L.polyline(coordinateArray, {color: '#f94000'}).addTo(_this.map);
+  flyToDevicePos: function() {
+    var _this = this;
 
-      console.log(coordinateArray);
+    _this.getDevicePos(function() {
+      _this.map.flyTo([_this.pos.lat, _this.pos.long], 20, {
+        animate: true
+      });
 
-      coordinateArray = [];
+      _this.stateDraw();
+      _this.bindStates();
+    });
+  },
 
-      navigator.geolocation.getCurrentPosition(function(position) {
-        _this.map.panTo(new L.LatLng(position.coords.latitude, position.coords.longitude));
-      });
-     
+  draw: function() {
+    var _this = this;
 
+    _this.map.on('mouseup', function() {
+      _this.map.off("mousemove");
+      L.polyline(_this.coordinateArray, {color: 'black'}).addTo(_this.map);
+      _this.coordinateArray = [];
     }).on('mousedown', function() {
-
       _this.map.on("mousemove", function(e) {
-        trackPoints(e)
+        _this.trackMouse(e);
       });
+    });
+  },
 
+  trackMouse: function(e) {
+    var _this = this;
+
+    _this.coordinateArray.push([e.latlng.lat, e.latlng.lng]);
+  },
+
+  stateDraw: function() {
```

Comment:
i think these functions need more comments to explain and also perhaps more descriptive names. is this setting a state or sorts?